subxt/src/backend/mod.rs

```diff
@@ -336,177 +336,308 @@ pub struct StorageResponse {
-                }
-                sub
+            pub async fn write(&self, items: Message<Item>) {
+                match items {
+                    Message::Many(items) => {
+                        for i in items.unwrap() {
+                            self.sender.send(i).await.unwrap()
+                        }
+                    }
+                    Message::Single(item) => self.sender.send(item).await.unwrap(),
+                };
             }
 
-            async fn read(&mut self) -> RpcResult<Vec<Item>> {
-                self.receiver.recv().await.unwrap()
+            pub async fn write_delayed(&self, items: Message<Item>) {
+                let sender = self.sender.clone();
+                tokio::spawn(async move {
+                    tokio::time::sleep(Duration::from_millis(500)).await;
+
+                    match items {
+                        Message::Many(items) => {
+                            for i in items.unwrap() {
+                                let _ = sender.send(i).await;
+                            }
+                        }
+                        Message::Single(item) => sender.send(item).await.unwrap(),
+                    };
+                });
             }
+        }
 
-            async fn write(&self, items: RpcResult<Vec<Item>>) {
-                self.sender.send(items).await.unwrap()
+        struct InnerMockedRpcClient {
+            data_table: MockDataTable,
+            subscription_channel: Option<Subscription>,
+            subscription_handlers: HashMap<String, SubscriptionHandler>,
+            method_handlers: HashMap<String, MethodHandler>,
+        }
+
+        impl InnerMockedRpcClient {
+            async fn call<'a>(
+                &'a mut self,
+                method_handler: &str,
+                params: Option<Box<serde_json::value::RawValue>>,
+            ) -> super::rpc::RawRpcFuture<'a, Box<serde_json::value::RawValue>> {
+                let method = self.method_handlers.get(method_handler).unwrap_or_else(|| {
+                    panic!(
+                        "no method named {} registered. Params: {:?}",
+                        method_handler, params
+                    )
+                });
+
+                (*method)(&mut self.data_table, &mut self.subscription_channel, params)
+            }
+
+            async fn subscribe<'a>(
+                &'a mut self,
+                sub: &str,
+                params: Option<Box<serde_json::value::RawValue>>,
+            ) -> super::rpc::RawRpcFuture<'a, super::rpc::RawRpcSubscription> {
+                let sub = self.subscription_handlers.get(sub).unwrap_or_else(|| {
+                    panic!(
+                        "no subscription named {} registered. Params: {:?}",
+                        sub, params
+                    )
+                });
+
+                (*sub)(&mut self.data_table, &mut self.subscription_channel, params)
             }
         }
+        pub struct MockRpcBuilder {
```

Comment:
can we `#[dervive(Default)]` here and remove the new impl below to save some lines of code?

Reply:
done